doc/langdef.md

```diff
@@ -1208,1337 +1207,938 @@ Regular expressions follow the
+
+```
+// true if the 'address' field exists in the 'user' message
+has(user.address)
+// true if map 'm' has a key named 'key_name' defined. The value may be null
+// as null does not connote absence in CEL.
+has(m.key_name)
+// false if the 'items' field is not set in the 'order' message
+has(order.items)
+// false if the 'user_id' key is not present in the 'sessions' map has(sessions.user_id)
+```
+
+**all \-** Tests whether all elements in the input list or all keys in a map
+satisfy the given predicate. The all macro behaves in a manner consistent with
+the Logical AND operator including in how it absorbs errors and short-circuits.
+
+**Signatures**
+
+*   `list(A).all(A, predicate(A) -> bool) -> bool`
+*   `map(A, B).all(A, predicate(A) -> bool) -> bool`
+
+**Examples**
+
+```
+[1, 2, 3].all(x, x > 0) // true
+[1, 2, 0].all(x, x > 0) // false
+['apple', 'banana', 'cherry'].all(fruit, fruit.size() > 3) // true
+[3.14, 2.71, 1.61].all(num, num < 3.0) // false
+{'a': 1, 'b': 2, 'c': 3}.all(key, key != 'b') // false
+```
+
+**exists** \- Tests whether any value in the list or any key in the map
+satisfies the predicate expression. The exists macro behaves in a manner
+consistent with the Logical OR operator including in how it absorbs errors and
+short-circuits.
+
+**Signatures**
+
+*   `list(A).exists(A, predicate(A) -> bool) -> bool`
+*   `map(A,B).exists(A, predicate(A) -> bool) -> bool`
+
+**Examples**
+
+```
+[1, 2, 3].exists(i, i % 2 != 0) // true
+[].exists(i, i > 0) // false
+[0, -1, 5].exists(num, num < 0) // true
+{'x': 'foo', 'y': 'bar'}.exists(key, key.startsWith('z')) // false
+```
+
+**exists\_one** \- Tests whether exactly one list element or map key satisfies
+the predicate expression. This macro does not short-circuit in order to remain
+consistent with logical operators being the only operators which can absorb
+errors within CEL.
+
+**Signatures**
+
+*   `list(A).exists_one(A, predicate(A)) -> bool`
+*   `map(A,B).exists_one(A, predicate(A)) -> bool`
+
+**Examples**
+
+```
+[1, 2, 2].exists_one(i, i < 2) // true
+{'a': 'hello', 'aa': 'hellohello'}.exists_one(k, k.startsWith('a')) // false
+[1, 2, 3, 4].exists_one(num, num % 2 == 0) // false
+```
+
+**filter** \- Returns a list containing only the elements from the input list
+that satisfy the given predicate
+
+**Signatures**
+
+*   `list(A).filter(A, function(A) -> bool) -> list(A)`
+*   `map(A, B).filter(A, function(A) -> bool) -> list(A)`
+
+**Examples**
+
+```
+[1, 2, 3].filter(x, x > 1) // [2, 3]
+['cat', 'dog', 'bird', 'fish'].filter(pet, pet.size() == 3) // ['cat', 'dog']
+[{'a': 10, 'b': 5, 'c': 20}].map(m, m.filter(key, m[key] > 10)) // [['c']]
+```
+
+**map** \- Returns a list where each element is the result of applying the
+transform expression to the corresponding input list element or input map key.
+
+There are both three argument and four argument forms of the macro. The three
+argument forms transform all elements. The four argument forms transform only
```

Comment:
grammar / clarity nit: pluralization is really odd in this description.

Something like:

For both maps and lists, the macro has a three argument and four argument form:
- The three argument form transforms all elements.
- The four argument form transforms only elements which satisfy the predicate. This is intended to simplify combined filter and map operations.